documentation/src/main/asciidoc/user_guide/chapter-76-Functional_Map_API.adoc

```diff
@@ -0,0 +1,775 @@
+import org.infinispan.commons.api.functional.FunctionalMap.*;
+import org.infinispan.commons.api.functional.Traversable;
+
+ReadOnlyMap<String, String> readOnlyMap = ...
+
+Set<String> keys = new HashSet<>();
+keys.add("key1");
+keys.add("key2");
+
+Traversable<String> values = readOnlyMap.evalMany(keys, ReadEntryView::get);
+values.forEach(System.out::println);
+----
+
+Finally, read-only map also exposes methods to read all existing keys as well
+as entries, which include both key and value information.
+
+[[_read_only_entry_view]]
+==== Read-Only Entry View
+
+The lambda parameters for read-only maps provide the user with a
+ link:https://docs.jboss.org/infinispan/8.0/apidocs/org/infinispan/commons/api/functional/EntryView.ReadEntryView.html[read-only entry view]
+ to interact with the data in the cache, which include these operations:
+
+* link:https://docs.jboss.org/infinispan/8.0/apidocs/org/infinispan/commons/api/functional/EntryView.ReadEntryView.html#key--[`key()`]
+ method returns the key for which this lambda is being executed.
+* link:https://docs.jboss.org/infinispan/8.0/apidocs/org/infinispan/commons/api/functional/EntryView.ReadEntryView.html#find--[`find()`]
+ returns a Java 8 `Optional` wrapping the value if present,
+ otherwise it returns an empty optional. Unless the value is guaranteed to
+ be associated with the key, it's recommended to use `find()` to verify
+ whether there's a value associated with the key.
+* link:https://docs.jboss.org/infinispan/8.0/apidocs/org/infinispan/commons/api/functional/EntryView.ReadEntryView.html#get--[`get()`]
+ returns the value associated with the key. If the key has no value
+ associated with it, calling `get()` throws a `NoSuchElementException`.
+ `get()` can be considered as a shortcut of `ReadEntryView.find().get()`
+ which should be used only when the caller has guarantees that there's
+ definitely a value associated with the key.
+* link:https://docs.jboss.org/infinispan/8.0/apidocs/org/infinispan/commons/api/functional/MetaParam.Lookup.html#findMetaParam-java.lang.Class-[`findMetaParam(Class<T> type)`]
+ allows metadata parameter information
+ associated with the cache entry to be looked up, for example: entry
+ lifespan, last  accessed time...etc.
+ See <<_meta_parameter, Metadata Parameter Handling section>> to find out more.
+
+=== Write-Only Map API
+
+Write-only operations include operations that insert or update data in the
+ cache and also removals. Crucially, a write-only operation does not attempt
+ to read any previous value associated with the key. This is an important
+ optimization since that means neither the cluster nor any persistence stores
+ will be looked up to retrieve previous values. In the main Infinispan Cache,
+ this kind of optimization was achieved using a local-only per-invocation
+ flag, but the use case is so common that in this new functional API, this
+ optimization is provided as a first-class citizen.
```

Comment:
The optimization is also available via `config.unsafe.unreliableReturnValues`...